src/lib/Lib/CMakeExecution.cpp

```diff
@@ -28,17 +29,70 @@ getCmakePath() {
     return *path;
 }
 
+Expected<std::string>
+executeCmakeHelp(llvm::StringRef cmakePath)
+{
+    llvm::SmallString<128> outputPath;
+    MRDOCS_CHECK(!llvm::sys::fs::createTemporaryFile("cmake-help", "txt", outputPath), 
+        "Failed to create temporary file");
+    std::optional<llvm::StringRef> const redirects[] = {llvm::StringRef(), outputPath.str(), llvm::StringRef()};
+    std::vector<llvm::StringRef> const args = {cmakePath, "--help"};
+    llvm::ArrayRef<llvm::StringRef> emptyEnv;
+    int const result = llvm::sys::ExecuteAndWait(cmakePath, args, emptyEnv, redirects);
+    MRDOCS_CHECK(result == 0, "CMake execution failed");
+
+    auto const bufferOrError = llvm::MemoryBuffer::getFile(outputPath);
+    MRDOCS_CHECK(bufferOrError, "Failed to read CMake help output");
+
+    return bufferOrError.get()->getBuffer().str();
+}
+
+Expected<std::string>
+getCmakeDefaultGenerator(llvm::StringRef cmakePath) 
+{
+    MRDOCS_TRY(auto const cmakeHelp, executeCmakeHelp(cmakePath));
+
+    std::istringstream stream(cmakeHelp);
+    std::string line;
+    std::string defaultGenerator;
+
+    while (std::getline(stream, line)) {
+        if (line[0] == '*' && line[1] == ' ') {
+            size_t const start = 2;
+            size_t const end = line.find("=", start);
+            if (end == std::string::npos) {
+                continue;
+            }
+            return line.substr(start, end - start);
+        }
+    }
+    return Unexpected(Error("Default CMake generator not found"));
+}
+
+Expected<bool>
+cmakeDefaultGeneratorIsVisualStudio(llvm::StringRef cmakePath) 
+{
+    MRDOCS_TRY(auto const defaultGenerator, getCmakeDefaultGenerator(cmakePath));
+    return defaultGenerator.find("Visual Studio") != std::string::npos;
+}
+
 Expected<std::string>
 executeCmakeExportCompileCommands(llvm::StringRef projectPath) 
 {
     MRDOCS_TRY(auto const cmakePath, getCmakePath());
     MRDOCS_CHECK(llvm::sys::fs::exists(projectPath), "CMakeLists.txt not found");
+    MRDOCS_TRY(auto const cmakeDefaultGeneratorIsVisualStudio, cmakeDefaultGeneratorIsVisualStudio(cmakePath));
 
     llvm::SmallString<128> tempDir;
     MRDOCS_CHECK(!llvm::sys::fs::createUniqueDirectory("compile_commands", tempDir), "Failed to create temporary directory");
 
     std::optional<llvm::StringRef> const redirects[] = {llvm::StringRef(), llvm::StringRef(), llvm::StringRef()};
-    std::vector<llvm::StringRef> const args = {cmakePath, "-S", projectPath, "-B", tempDir.str(), "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"};
+    std::vector<llvm::StringRef> args = {cmakePath, "-S", projectPath, "-B", tempDir.str(), "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"};
```

Comment:
-S and -B implies we require cmake >=3.13.5. 

When this feature is documented, this requirement needs to be included. 

I actually have implemented the logic to determine the cmake version and choose between the -S/-B and the older syntax in the cmake-workflows action but I don't think we need that here.